server/workflow/workflow_server.go

```diff
@@ -145,7 +145,7 @@ func (s *workflowServer) WatchWorkflows(req *workflowpkg.WatchWorkflowsRequest,
 		opts = req.ListOptions
 		wfName, err := argoutil.RecoverWorkflowNameFromSelectorString(opts.FieldSelector)
 		if err != nil {
-			return err
+			return fmt.Errorf("malformed request: workflows must be specified with a 'metadata.name=...' field selector; unable to parse: %s", err)
```

Comment:
minor - you should use `%w` rather than `%v` for errors, as that wraps the error rather that just the text: https://blog.golang.org/go1.13-errors

Reply:
Aside - I find that wrapping errors like this makes it SO easy to track errors to the line of code, I suggest we do this as a matter of course. @sarabala1979 @jessesuen @whynowy

Reply:
Will do